### ISSUE TITLE
Bug in writing incomplete Mark4 frame to file

### ISSUE THREAD

**Issue by cczhu** [mhvk/baseband#79]
Writing the sample Mark4 data to a new file, but not filling a frame leads to a ValueError.  For example, this code snippet: 

```
import baseband.vdif as vdif
import baseband.mark4 as mark4
from baseband.data import SAMPLE_MARK4
fr = mark4.open(SAMPLE_MARK4, 'rs', ntrack=64, decade=2010,
                frames_per_second=400)
record = fr.read(642)

with mark4.open("./my_partial_data.m4", 'ws', sample_rate=32*u.MHz,
                time=time0, ntrack=64, bps=2, fanout=4) as fw:
    # write in bits and pieces and with some invalid data as well.
    fw.write(record[:10])
    fw.write(record[10])
    fw.fh_raw.flush()
```
Leads to

```
ValueError: could not broadcast input array from shape (79989,8,8) into shape (79989,8)
```
following the warning that it will pad additional values to fill the frame.

The root cause is line 255 in vlbi_base/base.py, which assumes that channels are an separate dimension from thread (like for VDIF, where channel is "channels per frame").  Mark4StreamReader, though, inherits the line ```self.nthread = nchan if thread_ids is None else len(thread_ids)```.

**Comment by mhvk** [comment]
Good catch. I think the solution is to expose the `sample_shape` of the payload (or define it), and use that for creating the padding.

**Comment by mhvk** [comment]
fixed by #86
